src/workerd/util/sqlite.c++

```diff
@@ -30,6 +31,102 @@
 
 namespace workerd {
 
+namespace {
+
+// SQLite has a function like this in its internals, but it's not exposed to library consumers.
+//
+// These error codes come from https://www.sqlite.org/rescode.html#primary_result_code_list.
+kj::Maybe<kj::LiteralStringConst> errorName(int errorCode) {
+  switch (errorCode) {
+  case SQLITE_OK:
+    return "SQLITE_OK"_kjc;
+  case SQLITE_ERROR:
+    return "SQLITE_ERROR"_kjc;
+  case SQLITE_INTERNAL:
+    return "SQLITE_INTERNAL"_kjc;
+  case SQLITE_PERM:
+    return "SQLITE_PERM"_kjc;
+  case SQLITE_ABORT:
+    return "SQLITE_ABORT"_kjc;
+  case SQLITE_BUSY:
+    return "SQLITE_BUSY"_kjc;
+  case SQLITE_LOCKED:
+    return "SQLITE_LOCKED"_kjc;
+  case SQLITE_NOMEM:
+    return "SQLITE_NOMEM"_kjc;
+  case SQLITE_READONLY:
+    return "SQLITE_READONLY"_kjc;
+  case SQLITE_INTERRUPT:
+    return "SQLITE_INTERRUPT"_kjc;
+  case SQLITE_IOERR:
+    return "SQLITE_IOERR"_kjc;
+  case SQLITE_CORRUPT:
+    return "SQLITE_CORRUPT"_kjc;
+  case SQLITE_NOTFOUND:
+    return "SQLITE_NOTFOUND"_kjc;
+  case SQLITE_FULL:
+    return "SQLITE_FULL"_kjc;
+  case SQLITE_CANTOPEN:
+    return "SQLITE_CANTOPEN"_kjc;
+  case SQLITE_PROTOCOL:
+    return "SQLITE_PROTOCOL"_kjc;
+  case SQLITE_EMPTY:
+    return "SQLITE_EMPTY"_kjc;
+  case SQLITE_SCHEMA:
+    return "SQLITE_SCHEMA"_kjc;
+  case SQLITE_TOOBIG:
+    return "SQLITE_TOOBIG"_kjc;
+  case SQLITE_CONSTRAINT:
+    return "SQLITE_CONSTRAINT"_kjc;
+  case SQLITE_MISMATCH:
+    return "SQLITE_MISMATCH"_kjc;
+  case SQLITE_MISUSE:
+    return "SQLITE_MISUSE"_kjc;
+  case SQLITE_NOLFS:
+    return "SQLITE_NOLFS"_kjc;
+  case SQLITE_AUTH:
+    return "SQLITE_AUTH"_kjc;
+  case SQLITE_FORMAT:
+    return "SQLITE_FORMAT"_kjc;
+  case SQLITE_RANGE:
+    return "SQLITE_RANGE"_kjc;
+  case SQLITE_NOTADB:
+    return "SQLITE_NOTADB"_kjc;
+  case SQLITE_NOTICE:
+    return "SQLITE_NOTICE"_kjc;
+  case SQLITE_WARNING:
+    return "SQLITE_WARNING"_kjc;
+  case SQLITE_ROW:
+    return "SQLITE_ROW"_kjc;
+  case SQLITE_DONE:
+    return "SQLITE_DONE"_kjc;
+  default:
+    return kj::none;
+  }
+}
+
+kj::String namedErrorCode(int errorCode) {
```

Comment:
this can likely just return a kj::StringPtr instead

Reply:
It needs to dynamically build the result in the unknown-code case.